test/chainedvector.jl

```diff
@@ -369,6 +369,11 @@
 
     # https://github.com/JuliaData/CSV.jl/issues/842
     @test size(eachindex(ChainedVector([["a"]]))) == (1,)
+
+    x = [1, 2, 3]
+    append!(x, ChainedVector([[1, 2, 3]]))
+    @test length(x) == 6
+    @test x == [1, 2, 3, 1, 2, 3]
```

Comment:
Some more tests (some of them will fail now without the suggestions I make above):
```suggestion
    @test x == [1, 2, 3, 1, 2, 3]

    for (i, idx) in enumerate(eachindex(x))
       @test i == idx.i
    end

    i1 = first(eachindex(x))
    @test convert(Int,  i1) isa Int
    @test convert(Int8,  i1) isa Int8
    @test i1 == 1
    @test isequal(i1, 1)
    @test i1 < 2
    @test isless(i1, 2)
    @test i1 / 2 == 0.5
```